cmd/end2endtest/helpers.go

```diff
@@ -183,123 +181,65 @@ func (t *e2eElection) waitUntilElectionStarts(electionID types.HexBytes) (*vapi.
 	return election, nil
 }
 
-func (t *e2eElection) generateSIKProofs(root types.HexBytes) map[string]*apiclient.CensusProof {
-	type voterProof struct {
-		sikproof *apiclient.CensusProof
-		address  string
-	}
-	sikProofs := make(map[string]*apiclient.CensusProof, len(t.voterAccounts))
-	proofCh := make(chan *voterProof)
-	stopProofs := make(chan bool)
-	go func() {
-		for {
-			select {
-			case p := <-proofCh:
-				sikProofs[p.address] = p.sikproof
-			case <-stopProofs:
-				return
-			}
-		}
-	}()
-
+func (t *e2eElection) generateProofs(csp *ethereum.SignKeys, voterAccts []*ethereum.SignKeys) error {
+	var (
+		wg     sync.WaitGroup
+		vcount int32
+	)
+	errorChan := make(chan error, 1)
```

Comment:
this buffer with size 1 will deadlock as soon as 1 error is pushed into it
you want to `make(chan error, len(voterAccts)` instead

Reply:
or just `make(chan error)`, now that i saw the whole approach